pkg/providers/tinkerbell/stack/stack.go

```diff
@@ -0,0 +1,212 @@
+package stack
```

Comment:
Do we need a new package for this? This just seems like a component of the `tinkerbell` package.

Reply:
Packages can typically be argued any which way. I'm asking myself what is `stack` providing that I wouldn't expect from `tinkerbell`. With that in mind, I noticed `tinkerbell` imports `stack` to use some interfaces from it. At that point it started to feel like we've created a separation between 2 things that are working toward providing the same capability, hence should be in the same package (at least until we get all pieces in place and figure out a logical break of the `tinkerbell` package). 

Its trivial, for now, to change so you decide and we can rework as necessary after mulling it over for a few days.